plugins/outputs/pulsar/pulsar.go

```diff
@@ -0,0 +1,378 @@
+package pulsar
+
+import (
+	"context"
+	"fmt"
+	"github.com/gofrs/uuid"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/apache/pulsar-client-go/pulsar"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/influxdata/telegraf/plugins/serializers"
+)
+
+var ValidTopicSuffixMethods = []string{
+	"",
+	"measurement",
+	"tags",
+}
+
+var zeroTime = time.Unix(0, 0)
+
+type (
+	Pulsar struct {
+		URL             string `toml:"url"`
+		Topic           string `toml:"topic"`
+		TopicTag        string `toml:"topic_tag"`
+		ExcludeTopicTag bool   `toml:"exclude_topic_tag"`
+		RoutingTag      string `toml:"routing_tag"`
+		RoutingKey      string `toml:"routing_key"`
+
+		//Client Options
+		AuthProvider               string `toml:"auth_provider"`
+		TLSAllowInsecureConnection bool   `toml:"insecure_skip_verify"`
+		TLSTrustCertsFilePath      string `toml:"tls_ca"`
+		TLSValidateHostname        bool   `toml:"tls_validate_host_name"`
+		TLSCertificatePath         string `toml:"tls_cert"`
+		TLSPrivateKeyPath          string `toml:"tls_key"`
+		AuthToken                  string `toml:"auth_token"`
+		ConnectionTimeout          string `toml:"connection_timeout"`
+		OperationTimeout           string `toml:"operation_timeout"`
```

Comment:
Please using this
```suggestion
		ConnectionTimeout          internal.Duration `toml:"connection_timeout"`
		OperationTimeout           internal.Duration `toml:"operation_timeout"`
```
to allow the `toml` parser to parse the duration string for you.